pkg/disttask/framework/integrationtests/modify_test.go

```diff
@@ -0,0 +1,224 @@
+		testfailpoint.EnableCall(t, "github.com/pingcap/tidb/pkg/disttask/framework/scheduler/beforeRefreshTask", func(task *proto.Task) {
+			if task.State != proto.TaskStateRunning && task.Step != proto.StepTwo {
+				return
+			}
+			once.Do(func() {
+				require.NoError(t, c.TaskMgr.ModifyTaskByID(c.Ctx, task.ID, &proto.ModifyParam{
+					PrevState: proto.TaskStateRunning,
+					Modifications: []proto.Modification{
+						{Type: proto.ModifyConcurrency, To: 7},
+					},
+				}))
+				<-modifySyncCh
+			})
+		})
+		task, err := handle.SubmitTask(c.Ctx, "k2", proto.TaskTypeExample, 3, "", nil)
+		require.NoError(t, err)
+		require.Equal(t, 3, task.Concurrency)
+		// finish StepOne
+		subtaskCh <- struct{}{}
+		// wait task move to 'modifying' state
+		modifySyncCh <- struct{}{}
+		// wait task move back to 'running' state
+		require.Eventually(t, func() bool {
+			gotTask, err2 := c.TaskMgr.GetTaskByID(c.Ctx, task.ID)
+			require.NoError(t, err2)
+			return gotTask.State == proto.TaskStateRunning
+		}, 10*time.Second, 100*time.Millisecond)
+		// finish StepTwo
+		subtaskCh <- struct{}{}
+		task2Base := testutil.WaitTaskDone(c.Ctx, t, task.Key)
+		require.Equal(t, proto.TaskStateSucceed, task2Base.State)
+		checkSubtaskConcurrency(t, c, task.ID, map[proto.Step]int{
+			proto.StepOne: 3,
+			proto.StepTwo: 7,
+		})
+	})
+
+	t.Run("modify paused task concurrency", func(t *testing.T) {
+		var once sync.Once
+		syncCh := make(chan struct{})
+		var theTask *proto.Task
+		testfailpoint.EnableCall(t, "github.com/pingcap/tidb/pkg/disttask/framework/scheduler/beforeGetSchedulableTasks", func() {
+			once.Do(func() {
+				task, err := handle.SubmitTask(c.Ctx, "k3", proto.TaskTypeExample, 3, "", nil)
+				require.NoError(t, err)
+				require.Equal(t, 3, task.Concurrency)
+				found, err := c.TaskMgr.PauseTask(c.Ctx, task.Key)
+				require.NoError(t, err)
+				require.True(t, found)
+				theTask = task
+				<-syncCh
+			})
+		})
+		syncCh <- struct{}{}
+		taskBase := testutil.WaitTaskDoneOrPaused(c.Ctx, t, theTask.Key)
+		require.Equal(t, proto.TaskStatePaused, taskBase.State)
+		require.NoError(t, c.TaskMgr.ModifyTaskByID(c.Ctx, theTask.ID, &proto.ModifyParam{
+			PrevState: proto.TaskStatePaused,
+			Modifications: []proto.Modification{
+				{Type: proto.ModifyConcurrency, To: 7},
+			},
+		}))
+		taskBase = testutil.WaitTaskDoneOrPaused(c.Ctx, t, theTask.Key)
+		require.Equal(t, proto.TaskStatePaused, taskBase.State)
+		found, err := c.TaskMgr.ResumeTask(c.Ctx, theTask.Key)
+		require.NoError(t, err)
+		require.True(t, found)
+		// finish subtasks
+		subtaskCh <- struct{}{}
+		subtaskCh <- struct{}{}
+		task2Base := testutil.WaitTaskDone(c.Ctx, t, theTask.Key)
+		require.Equal(t, proto.TaskStateSucceed, task2Base.State)
+		checkSubtaskConcurrency(t, c, theTask.ID, map[proto.Step]int{
+			proto.StepOne: 7,
+			proto.StepTwo: 7,
+		})
+	})
+
+	t.Run("modify pending task concurrency, but other owner already done it", func(t *testing.T) {
```

Comment:
The test purpose is "..but other owner already done it". Should we trigger ModifiedTask after WaitTaskDone and check the concurrency is not changed?

Reply:
> Should we trigger ModifiedTask after WaitTaskDone

the scenario is 
- task change to `modifying` state
- owner A tried to do modify work, but before it start, another owner B finished the modify
- owner A does nothing

> check the concurrency is not changed?

on owner A, we check by `AffectedRows() == 0`